sql-cli/sql_cli/__main__.py

```diff
@@ -29,16 +28,38 @@
     rich_markup_mode="rich",
 )
 app.add_typer(cli_config.app, name="config", hidden=True)
+log = logging.getLogger("sql_cli")
 
-airflow_logger = logging.getLogger("airflow")
-airflow_logger.setLevel(logging.CRITICAL)
-airflow_logger.propagate = False
 
-boto_logger = logging.getLogger("botocore")
-boto_logger.setLevel(logging.CRITICAL)
+def set_verbose_mode(verbose: bool) -> None:
+    """
+    Sets the logging level for the sql-cli to be more verbose.
+
+    :param verbose: Whether verbose logs should be enabled.
+    """
+    log.propagate = False
+    log.addHandler(RichHandler(markup=True))
+    if verbose:
+        log.setLevel(logging.DEBUG)
+    else:
+        log.setLevel(logging.CRITICAL)
+
+
+def set_debug_mode(debug: bool) -> None:
+    """
+    Sets the logging level for all external dependencies.
+
+    :param debug: Whether debug logs should be enabled.
+    """
+    STATE["debug"] = debug
 
-snowflake_logger = logging.getLogger("snowflake")
-snowflake_logger.setLevel(logging.CRITICAL)
+    for logger_name in ["airflow", "botocore", "snowflake"]:
```

Comment:
It's probably worth having a constant with the values of the loggers we want to handle this way